tests/common/type_coverage.h

```diff
@@ -231,4 +230,49 @@ void for_all_types_vectors_marray(const named_type_pack<types...>& typeList,
       )...};
   static_cast<void>(packExpansion);
 }
+
+/**
+ * @brief Run action for type and marrays of this type
+ * @tparam action Functor template for action to run
+ * @tparam T Type to instantiate functor template with
+ * @tparam actionArgsT Parameter pack to use for functor template instantiation
+ * @tparam argsT Deduced parameter pack for arguments to forward into the call
+ * @param args Arguments to forward into the call
+ */
+template <template <typename, typename...> class action, typename T,
+          typename... actionArgsT, typename... argsT>
+void for_type_and_marrays(argsT&&... args) {
+  for_all_types<action, actionArgsT...>(
+      type_pack<T, typename sycl::template marray<T, 2>,
+                typename sycl::template marray<T, 5>,
+                typename sycl::template marray<T, 10>>{},
+      std::forward<argsT>(args)...);
+}
+
+/**
+ * @brief Run action for each of types and marrays of types given by
+ *        named_type_pack instance
+ * @tparam action Functor template for action to run
+ * @tparam actionArgsT Parameter pack to use for functor template instantiation
+ * @tparam types Deduced from type_pack parameter pack for list of types to use
+ * @tparam argsT Deduced parameter pack for arguments to forward into the call
+ * @param typeList Named type pack instance with underlying type names stored
+ * @param args Arguments to forward into the call
+ */
+template <template <typename, typename...> class action,
+          typename... actionArgsT, typename... types, typename... argsT>
+void for_all_types_and_marrays(const named_type_pack<types...>& typeList,
+                               argsT&&... args) {
+  /** run action for each type from types... parameter pack
+   */
+  size_t typeNameIndex = 0;
+
+  int packExpansion[] = {(
+      for_type_and_marrays<action, types, actionArgsT...>(
+          std::forward<argsT>(args)..., typeList.names[typeNameIndex]),
+      ++typeNameIndex,
+      0  // Dummy initialization value
+      )...};
+  static_cast<void>(packExpansion);
```

Comment:
Just back from CppCon, it looks like I have seen simpler way like:
```c++
#include <iostream>
void print(auto a, int i) {
    std::cout << i << ':' << a << std::endl;
}
void expand(auto... a) {
    int i = 0;
    ((print(a, i), i++),...);
}

int main() {
    expand(1,2,3);
}
```
https://godbolt.org/z/qTW1r79We

So I have the feeling the following is enough
```suggestion
  ((for_type_and_marrays<action, types, actionArgsT...>(
          std::forward<argsT>(args)..., typeList.names[typeNameIndex]),
    ++typeNameIndex), ...);
```

Reply:
this commend was applied [here](https://github.com/KhronosGroup/SYCL-CTS/pull/202/commits/7b8b881e2665a5081119f502def79cecae04f2c3#diff-6c3f2e5414e9e68ce7030404ce12e917db441dd8be63f28613a6b951faf88c8bL270-L277)